app/build/app/js/app.js

```diff
@@ -70,6 +72,21 @@ angular.module('media_manager').controller('CollectionsController', [
         lc.deleteCollectionModal = CollectionBehavior.deleteCollectionModal;
         lc.actuallyDeleteCollection = CollectionBehavior.actuallyDeleteCollection;
         lc.isLoadingCollections = CourseCache.isLoadingCollections;
+
+        lc.dragControlListeners = {
+          orderChanged: function(event){
+
+            lc.collections.forEach(function(item, index, arr){
+              var newsort = index + 1;
+              if(item.sort_order !== newsort){
+                arr[index].sort_order = newsort;
+                Collection.update({id: item.id}, arr[index]);
```

Comment:
So each `Collection.update()` will trigger an AJAX request to the API to update the collection's sort order. We could end up updating every collection if the user drags the last collection into the first position. Are we doing anything to make sure that the user can't kick off another set of updates until the first batch of updates has completed?

Example scenario:

1) There are 6 collections. The user drags the last collection into the first position, therefore triggering 6 AJAX requests to update them.
2) Before those 6 AJAX requests are completed, the user does the same thing again on the UI (drag&drop last collection to the first position), triggering yet another 6 AJAX requests.

What is the final sort order of the collections when those 12 AJAX requests are done? I can't predict what that sort order will be because those requests are asynchronous and I can't say with any certainty that the first batch of requests will be processed before the second batch. 

With this kind of approach, I think we'd need to setup some kind of queue and track each batch of updates with a promise or something so that we could use to guarantee that a subsequent drag&drop wouldn't be processed before the previous one. 

But now, things are getting rather complicated with queues and multiple promises per reorder. It seems to me that we really would be better off if we only had to issue *one* request per reordering, since the queue would just be an array of promises, each promise representing one reorder request. That would also have the side benefit of scaling nicely since it wouldn't matter how many collections were being reordered.